aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -348,49 +723,77 @@ module aptos_framework::coin {
-    ) acquires CoinInfo, CoinStore {
+    ) acquires CoinInfo, CoinStore, CoinConversionMap {
         // Skip burning if amount is zero. This shouldn't error out as it's called as part of transaction fee burning.
         if (amount == 0) {
             return
         };
 
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        let coin_to_burn = extract(&mut coin_store.coin, amount);
-        burn(coin_to_burn, burn_cap);
+        let (coin_amount_to_burn, fa_amount_to_burn) = calculate_amount_to_withdraw<CoinType>(
+            account_addr,
+            amount
+        );
+        if (coin_amount_to_burn > 0) {
+            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
+            let coin_to_burn = extract(&mut coin_store.coin, coin_amount_to_burn);
+            burn(coin_to_burn, burn_cap);
+        };
+        if (fa_amount_to_burn > 0) {
+            let store_addr = primary_fungible_store::primary_store_address(
+                account_addr,
+                option::destroy_some(paired_metadata<CoinType>())
+            );
+            let fa = fungible_asset::withdraw_internal(store_addr, fa_amount_to_burn);
+            fungible_asset::burn_internal(fa);
+        };
     }
 
     /// Deposit the coin balance into the recipient's account and emit an event.
-    public fun deposit<CoinType>(account_addr: address, coin: Coin<CoinType>) acquires CoinStore {
-        assert!(
-            is_account_registered<CoinType>(account_addr),
-            error::not_found(ECOIN_STORE_NOT_PUBLISHED),
-        );
-
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        assert!(
-            !coin_store.frozen,
-            error::permission_denied(EFROZEN),
-        );
-        event::emit_event<DepositEvent>(
-            &mut coin_store.deposit_events,
-            DepositEvent { amount: coin.value },
-        );
-        event::emit(Deposit<CoinType> { account: account_addr, amount: coin.value });
-
-        merge(&mut coin_store.coin, coin);
+    public fun deposit<CoinType>(
+        account_addr: address,
+        coin: Coin<CoinType>
+    ) acquires CoinStore, CoinConversionMap, CoinInfo {
+        if (exists<CoinStore<CoinType>>(account_addr)) {
+            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
+            assert!(
+                !coin_store.frozen,
+                error::permission_denied(EFROZEN),
+            );
+            event::emit_event<DepositEvent>(
+                &mut coin_store.deposit_events,
+                DepositEvent { amount: coin.value },
+            );
+            event::emit(Deposit<CoinType> { account: account_addr, amount: coin.value });
+
+            merge(&mut coin_store.coin, coin);
+        } else if (features::coin_to_fungible_asset_migration_feature_enabled()) {
+            primary_fungible_store::deposit(account_addr, coin_to_fungible_asset(coin));
```

Comment:
wait, I thought this wasn't a coerced upgrade... this means if I send apt to a new account in a CEX it won't be picked up until they support fa...

Reply:
This one is debatable... 
Since I can send you FA via `primary_fungible_store::transfer`. Do we want to disable the `coin::transfer` to PFS?